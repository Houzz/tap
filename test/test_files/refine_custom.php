<?php

class ClassA {
    public function foo() {}
};

/**
 * @param ClassA|null $p1
 */
function refineCustomClass($p1) {
    $p1->foo();     // error
    if ($p1 !== null) {
        $p1->foo();
        $p1->bar(); // error
    }
    $p1->foo();     // error
    if ($p1 === null)
        return;
    $p1->foo();
}

