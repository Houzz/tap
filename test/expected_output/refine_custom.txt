11 E_UNION_TYPE_METHOD
14 E_METHOD_NOT_FOUND
16 E_UNION_TYPE_METHOD
